general/src/store/normalize.rs

```diff
@@ -232,28 +232,37 @@ impl<'a> NormalizeProcessor<'a> {
 
     /// Inserts the IP address into the user interface. Creates the user if it doesn't exist.
     fn normalize_user_ip(&self, event: &mut Event) {
-        let remote_addr = event
+        // If there is no User ip_address, update it either from the Http interface
+        // or the client_ip of the request.
+        let ip_address = event
             .request
             .value()
             .and_then(|request| request.env.value())
             .and_then(|env| env.get("REMOTE_ADDR"))
             .and_then(Annotated::<Value>::as_str)
-            .and_then(|ip| IpAddr::parse(ip).ok());
-
-        // If there is no User ip_address, update it either from the Http interface
-        // or the client_ip of the request.
-        if let Some(ip_address) = remote_addr {
-            self.set_user_ip(&mut event.user, ip_address);
-        } else if let Some(ref ip_address) = self.config.client_ip {
-            self.set_user_ip(&mut event.user, ip_address.clone());
-        }
-    }
-
-    /// Creates or updates the user's IP address.
-    fn set_user_ip(&self, user: &mut Annotated<User>, ip_address: IpAddr) {
-        let user = user.value_mut().get_or_insert_with(User::default);
-        if user.ip_address.value().is_none() {
-            user.ip_address = Annotated::new(ip_address);
+            .and_then(|ip| IpAddr::parse(ip).ok())
+            .or_else(|| self.config.client_ip.clone());
+
+        if let Some(ip_address) = ip_address {
+            let platform = event.platform.value().map(String::as_str);
```

Comment:
_nit_: Inline this into the match statement.

```suggestion
            let platform = event.platform.as_str();
```